gateway-js/src/loadCsdlFromStorage.ts

```diff
@@ -0,0 +1,133 @@
+import { fetch } from 'apollo-server-env';
+import { GraphQLError } from 'graphql';
+
+export const CSDL_QUERY = `#graphql
+  query Csdl($apiKey: String!, $ref: String!) {
+    routerConfig(
+      ref: $ref
+      apiKey: $apiKey
+    ) {
+      __typename
+      ... on RouterConfigResult {
+        id
+        csdl
+      }
+      ... on FetchError {
+        code
+        message
+      }
+    }
+  }
+`;
+
+type CsdlQueryResult = CsdlQuerySuccess | CsdlQueryFailure;
+
+interface CsdlQuerySuccess {
```

Comment:
Seems like we could use codegen here? (Probably graphql-code-generator) OK if that's overkill though

Reply:
Like this idea and always wanted a reason to try out that tool. It was a great experience!
9706663